src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java

```diff
@@ -1001,9 +1001,18 @@ private static Configuration dataFileConfiguration(Project project, String name,
             spec.withDependencies(dependencies -> dependencies.add(depFactory.create(collection.getPublished())));
         });
 
-        // Set up the publishing conditionally
-        AdhocComponentWithVariants java = (AdhocComponentWithVariants) project.getComponents().getByName("java");
+        // Publish the data files as artifacts whose classifiers match the name of the file to avoid conflicts
+        // Unfortunately, ArtifactHandler does not have lazy methods, so afterEval it is
+        project.afterEvaluate(proj -> {
+            for (var file : collection.getPublished()) {
+                proj.getArtifacts().add(elementsConfiguration.getName(), file, configurablePublishArtifact -> {
+                    configurablePublishArtifact.setClassifier(FileUtils.stripExtension(file.getName()));
```

Comment:
I suppose that Gradle already finds out about the extension? What is the default classifier that it chooses then? (i.e. Why do we need to override the classifier and strip the extension?).

Reply:
It doesn't chose anything, for the subproject with an iface injection data file for instance, the artifact is `subproject-<version>.json`.